### PR TITLE
bugfix/AB#81961_ABC-issue-with-grid-actions-tab-in-summary-card-settings 

### PR DESCRIPTION
**Pull request by unai-reliefapp**
# Description

fix: show grid actions tab on resource and layout set or when grid display option is enabled in display settings tab

## Useful links

- [Link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/81961)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please refer to video below

## Screenshots

https://github.com/ReliefApplications/ems-frontend/assets/123092672/6220d66b-3776-4c1c-94fb-501648c3563e

# Checklist:

( * == Mandatory ) 

- [x] * I have set myself as assignee of the pull request
- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

**Review by guisoares1 — CHANGES_REQUESTED** [review]
@unai-reliefapp the initial problem is no longer presenting. 
The only detail that I believe we can improve is to continue focusing on the 'Widget display' tab after changing the 'Allow to show as grid' option.

**Comment by unai-reliefapp** [comment]
> @unai-reliefapp the initial problem is no longer presenting. The only detail that I believe we can improve is to continue focusing on the 'Widget display' tab after changing the 'Allow to show as grid' option.

Updated PR with the fix @guisoares1 , thanks for the spot. I rather keep the tab auto select logic in the summary card settings instead of in the tabs component as that would affect all the components in the application.

Fixed behavior:

https://github.com/ReliefApplications/ems-frontend/assets/123092672/4079927e-a8fd-41db-ab5c-a1fde8c70317

**Comment by AntoineRelief** [comment]
@unai-reliefapp @guisoares1 
I'll take the final review, thanks both

File: libs/shared/src/lib/components/widgets/summary-card-settings/summary-card-settings.component.html [hunk 1]
```diff
@@ -54,7 +58,12 @@
       </ng-template>
     </ui-tab>
     <!-- Available actions -->
-    <ui-tab *ngIf="layout">
+    <ui-tab
+      *ngIf="
+        (resource && layout) ||
+        $any(widgetFormGroup.get('widgetDisplay.gridMode'))?.value
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/widgets/summary-card-settings/summary-card-settings.component.html:65]: @unai-reliefapp I think it should be a && there

Comment (unai-reliefapp) [reply libs/shared/src/lib/components/widgets/summary-card-settings/summary-card-settings.component.html:65]: okay, I understood this as two different conditions
image
would you update it or should i do it?

Comment (AntoineRelief) [reply libs/shared/src/lib/components/widgets/summary-card-settings/summary-card-settings.component.html:65]: just did 👍

**Pull request merged**